docs/docsite/rst/dev_guide/developing_plugins.rst

```diff
@@ -36,6 +36,17 @@ You should return errors encountered during plugin execution by raising ``Ansibl
     except Exception as e:
         raise AnsibleError('Something happened, this was original exception: %s' % to_native(e))
 
+Since Ansible evaluates variables only when they are needed, filter and test plugins should propagate the exceptions ``jinja2.exceptions.UndefinedError`` and ``AnsibleUndefinedVariable`` to ensure undefined variables are only fatal when necessary.
```

Comment:
Does this only apply to filter/test plugins? If so, I'd move this information into the sections on those specific plugins.

Reply:
Yes, it does. How does 039440f4af178e74e78b66e86f2622ed05b58d44 look?